test/distributed/test_compute_comm_reordering.py

```diff
@@ -0,0 +1,250 @@
+            ar = _functional_collectives.all_reduce(a, "sum", ranks, tag)
+            g = torch.matmul(a, a)
+            c = torch.relu(a)
+            d = torch.matmul(c, c)
+            f = d * c * ar
+            fr = _functional_collectives.all_reduce(f, "sum", ranks, tag)
+            e = torch.matmul(d + ar + fr, g)
+            return (e,)
+
+        with _dynamo_dist_per_rank_init(self.rank, self.world_size):
+            inputs = torch.ones(4, 4, dtype=torch.float, device="cuda") + self.rank
+            compiled = torch.compile(func)
+            code = run_and_get_triton_code(compiled, inputs, **self.get_world_trs())
+            # NOTE: after scheduling the first all_reduce:
+            # 1. we first schedule the ops (c and d) that ARE required for second all_reduce but DO NOT depend on first all_reduce.
+            # 2. then, we schedule the ops (g) that ARE NOT required for second all_reduce and DO NOT depend on first all_reduce.
+            # 3. then, we schedule the ops (f) that ARE required for second all_reduce and DO depend on first all_reduce.
+            # and then, we schedule the second all_reduce. And then schedule all ops that depend on second all_reduce.
+            FileCheck() \
+                .check("dist.all_reduce(") \
+                .check("triton_poi_fused_relu") \
+                .check("extern_kernels.mm(") \
+                .check("extern_kernels.mm(") \
+                .check("_wait_tensor(") \
+                .check("triton_poi_fused_mul") \
+                .check("dist.all_reduce(") \
+                .check("_wait_tensor(") \
+                .check("triton_poi_fused_add") \
+                .check("extern_kernels.mm(") \
+                .run(code)
+            out = compiled(inputs, **self.get_world_trs())
+            correct = func(inputs, **self.get_world_trs())
+            self.assertTrue(same(out, correct))
+
+    def get_snode_runtime_for_reorder_compute_test(snode):
+        # NOTE: custom cost model to show that the compute reordering algorithm is working
+        # Collective kernels
+        if isinstance(snode.node, ir.CollectiveKernel):
+            if isinstance(snode.node, ir.AllReduce):
+                return 100
+            else:
+                return 100
+        elif isinstance(snode.node, ir.Wait):
+            return 0
+        # High-arithmetic-intensity compute kernels
+        elif isinstance(snode.node, ir.ExternKernel):
+            return 5
+        # All other kernels
+        return 1
+
+    @unittest.skipIf(not has_triton(), "Inductor+gpu needs triton and recent GPU arch")
+    @skip_if_lt_x_gpu(2)
+    @patch.object(torch._inductor.config, "allow_buffer_reuse", True)
+    # TODO: somehow inductor bg compile threads are causing hangs at exit with distributed work dtor
+    @patch.object(torch._inductor.config, "compile_threads", 1)
+    @patch.object(torch._inductor.config, "reorder_for_compute_comm_overlap", True)
```

Comment:
for usage, we can simply do torch._inductor.config.reorder_for_compute_comm_overlap = True and no need to patch the passes right?

Reply:
yes that's right